// @webpack-cli/generators/lib/utils/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validate = exports.tooltip = exports.StyleRegex = exports.LoaderName = exports.StylingType = exports.styleQuestionHandler = exports.getTypescriptLoader = exports.getBabelLoader = exports.LangType = exports.langQuestionHandler = exports.entryQuestions = void 0;
const entry_1 = __importDefault(require("./entry"));
exports.entryQuestions = entry_1.default;
const languageSupport_1 = __importStar(require("./languageSupport"));
exports.langQuestionHandler = languageSupport_1.default;
Object.defineProperty(exports, "LangType", { enumerable: true, get: function () { return languageSupport_1.LangType; } });
Object.defineProperty(exports, "getBabelLoader", { enumerable: true, get: function () { return languageSupport_1.getBabelLoader; } });
Object.defineProperty(exports, "getTypescriptLoader", { enumerable: true, get: function () { return languageSupport_1.getTypescriptLoader; } });
const styleSupport_1 = __importStar(require("./styleSupport"));
exports.styleQuestionHandler = styleSupport_1.default;
Object.defineProperty(exports, "StylingType", { enumerable: true, get: function () { return styleSupport_1.StylingType; } });
Object.defineProperty(exports, "LoaderName", { enumerable: true, get: function () { return styleSupport_1.LoaderName; } });
Object.defineProperty(exports, "StyleRegex", { enumerable: true, get: function () { return styleSupport_1.StyleRegex; } });
const tooltip_1 = __importDefault(require("./tooltip"));
exports.tooltip = tooltip_1.default;
const validate_1 = __importDefault(require("./validate"));
exports.validate = validate_1.default;
